Added - Font Awesome Icon Picker (PRESIDECMS-266)

josephpbuarao opened the pull request:
Hi @DominicWatson ,

Good day! I've added a new form control "iconpicker". Please take a look if you have time.

Thank you and have a great day ahead.

josephpbuarao committed:
Added - Font Awesome Icon Picker (PRESIDECMS-266)

josephpbuarao committed:
Changed icon picker position

DominicWatson commented on system/assets/js/admin/presidecore/formFields.js:
```diff
@@ -105,4 +105,10 @@
 
+
+    $('.icon-picker-fontawesome').iconpicker({
+        title: 'Using Font Awesome',
```
This can/should use:

```
title : i18n.translateResource( 'cms:font.awesome.picker.title' )
```

You can then add corresponding i18n .properties entry.

DominicWatson commented on system/views/formcontrols/iconpicker/index.cfm:
```diff
@@ -0,0 +1,23 @@
+		value = "";
+	}
+
+	if ( IsDate( value ) ) {
```
I think we shouldn't be looking for dates here? Perhaps this has been left after copy and paste?

DominicWatson commented on system/assets/js/admin/lib/plugins/jquery.fontawesome-iconpicker.js:
```diff
@@ -0,0 +1,995 @@
+    "use strict";
+    if (typeof define === "function" && define.amd) {
+        define([ "jquery" ], a);
+    } else if (window.jQuery && !window.jQuery.fn.iconpicker) {
```
I think all references to `window.jQuery` should be replaced with `window.presideJQuery` here

josephpbuarao committed:
Added: hideOnSelect: true
Added: i18n property for icon picker
Removed: unnecessary code in form control icon picker
Replaced: window.jQuery to window.presideJQuery

josephpbuarao commented:
Hi @DominicWatson ,

Thank you for your time reviewing my pull request. I already implemented what you have suggested.

Thank you very much!

DominicWatson commented:
Nice one @josephpbuarao . Merged into upcoming 10.5.0 release :)
